Mathlib/CategoryTheory/Galois/Basic.lean

```diff
@@ -123,8 +127,77 @@ noncomputable instance : ReflectsColimitsOfShape (Discrete PEmpty.{1}) F :=
 noncomputable instance : PreservesFiniteLimits F :=
   preservesFiniteLimitsOfPreservesTerminalAndPullbacks F
 
+/-- Fibre functors reflect monomorphisms. -/
+instance foo : ReflectsMonomorphisms F := ReflectsMonomorphisms.mk <| by
+  intro X Y f _
+  haveI : IsIso (pullback.fst : pullback (F.map f) (F.map f) ⟶ F.obj X) :=
+    fst_iso_of_mono_eq (F.map f)
+  haveI : IsIso (F.map (pullback.fst : pullback f f ⟶ X)) := by
+    rw [← PreservesPullback.iso_hom_fst]
+    exact IsIso.comp_isIso
+  haveI : IsIso (pullback.fst : pullback f f ⟶ X) := isIso_of_reflects_iso pullback.fst F
+  exact (pullback.diagonal_isKernelPair f).mono_of_isIso_fst
+
+/-- Fibre functors are faithful. -/
+instance : Faithful F where
+  map_injective {X Y} f g h := by
+    haveI : IsIso (equalizer.ι (F.map f) (F.map g)) := equalizer.ι_of_eq h
+    haveI : IsIso (F.map (equalizer.ι f g)) := by
+      rw [← equalizerComparison_comp_π f g F]
+      exact IsIso.comp_isIso
+    haveI : IsIso (equalizer.ι f g) := isIso_of_reflects_iso _ F
+    exact eq_of_epi_equalizer
+
 end FibreFunctor
 
+variable (F : C ⥤ FintypeCat.{w}) [FibreFunctor F]
+
+private lemma FintypeCat.initial_iff_empty (X : FintypeCat.{w}) :
```

Comment:
Can this go a more basic file? And I think there is no need to have it private.

Reply:
See the discussion above. Should I just make a new file `Mathlib.CategoryTheory.Limits.Shapes.FintypeCat`? Or do you have a different suggestion? Maybe this can be generalized to concrete categories somehow?